src/warc2zim/content_rewriting/generic.py

```diff
@@ -94,6 +96,8 @@ def rewrite(
             return self.rewrite_css()
 
         if self.rewrite_mode == "javascript":
+            if any(path.value == self.path for path in self.js_modules):
```

Comment:
Could we implement a `__eq__` on `ZimPath` to avoid this `path.value` ?

Reply:
We could, but I don't want to ^^

Being forced to do it is exactly the purpose of this class (and the `HttpUrl`) so that we make it intentional to compare a `ZimPath` with a string (should not be the normal case).

Actually here the mistake is that I should pass a `ZimPath` to the rewriters to make things even clearer. But this is beyond the scope of this PR, I will move this to a dedicated issue: https://github.com/openzim/warc2zim/issues/236